### DISCUSSION TITLE
Serializer with InputStream/OutputStream

### DISCUSSION THREAD

**Discussion by lucasmogari** [microstream-one/microstream#623] [Q&A]
Is it possible to use serializer with InputStream and OutputStream without and intermediary byte array?

For example, read/write an instance object (not in microstream graph) to a file.

**Comment by fh-ms** [comment]
Of course, just write the bytes produced by the serializer into the stream, and vice versa.

```java
Serializer<byte[]> serializer = Serializer.Bytes();
byte[]             bytes      = serializer.serialize("Hello!");
try(OutputStream out = new FileOutputStream(myFile))
{
	out.write(bytes);
}
catch(IOException e)
{
	throw new RuntimeException(e);
}
```

**Reply by lucasmogari** [reply]
Sorry, I forgot to mention that I didn't want to load the whole file in memory because it'll throw an OOME. Is it possible?

**Reply by fh-ms** [reply]
If you create objects out of the file's data, the memory will be consumed anyway. So better increase the heap size.

**Reply by lucasmogari** [reply]
I'd like to avoid loading the full file byte array and the object instance in memory at once. Sometimes, the file is big, and it's not possible to increase the heap.

Kryo has an interface for InputStream and can stream the file while deserializing. That's why I was curious to see if microstream has something like that.

**Reply by lucasmogari** [reply]
I tried the solution with the byte[] and it throws a NegativeArraySizeException. I think the object is too big.

**Reply by fh-ms** [reply]
Ah, I see. In that case, this would come in useful. I opened an issue (#624)

**Reply by lucasmogari** [reply]
Thank you!
